workspaces/kiali/plugins/kiali-backend/src/clients/fetch.ts

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { LoggerService } from '@backstage/backend-plugin-api';
+
+import axios, { AxiosError, AxiosRequestConfig } from 'axios';
+
+import fs from 'fs';
+import https from 'https';
+
+import { KialiDetails } from '../service/config';
+import {
+  AuthInfo,
+  AuthStrategy,
+  KialiAuthentication,
+  SessionInfo,
+} from './Auth';
+
+export enum ValidationCategory {
+  configuration = 'configuration',
+  authentication = 'authentication',
+  versionSupported = 'versionSupported',
+  networking = 'networking',
+  unknown = 'unknown',
+}
+
+export type KialiValidations = {
+  verify: boolean;
+  category: ValidationCategory;
+  title?: string;
+  missingAttributes?: string[];
+  message?: string;
+  helper?: string;
+  authData?: AuthInfo;
+};
+
+const TIMEOUT_FETCH = 8000;
+export class KialiFetcher {
+  private readonly logger: LoggerService;
+  private kialiAuth: KialiAuthentication;
+  private KialiDetails: KialiDetails;
+
+  constructor(KD: KialiDetails, logger: LoggerService) {
+    this.KialiDetails = KD;
+    this.logger = logger;
+    this.kialiAuth = new KialiAuthentication(KD);
+  }
+
+  newRequest = async <P>(endpoint: string, auth: boolean = false) => {
+    this.logger.info(`Query to ${endpoint}`);
+    return axios.request<P>(this.getRequestInit(endpoint, auth));
+  };
+
+  private async getAuthInfo(): Promise<AuthInfo> {
+    return this.newRequest<AuthInfo>('api/auth/info').then(resp => resp.data);
+  }
+
+  getAuthData(): AuthInfo {
+    return this.kialiAuth.getSession();
+  }
+
+  private validateConfiguration = (auth: AuthInfo): KialiValidations => {
+    const result: KialiValidations = {
+      verify: true,
+      category: ValidationCategory.unknown,
+      authData: auth,
+    };
+    switch (auth.strategy) {
+      case AuthStrategy.anonymous:
+        break;
+      case AuthStrategy.token: {
+        if (
+          !this.KialiDetails.serviceAccountToken ||
+          this.KialiDetails.serviceAccountToken === ''
+        ) {
+          result.verify = false;
+          result.title = 'Authentication failed. Missing Configuration';
+          result.category = ValidationCategory.configuration;
+          result.message = `Attribute 'serviceAccountToken' is not in the backstage configuration`;
+          result.helper = `For more information follow the steps in https://janus-idp.io/plugins/kiali`;
```

Comment:
Since we are sunsetting janus, we should avoid redirecting users to https://janus-idp.io/